flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/typeutils/RowDataSerializerTest.java

```diff
@@ -205,44 +243,77 @@ private static GenericRowData createRow(Object f0, Object f1, Object f2, Object
 	}
 
 	private static boolean deepEqualsRowData(
-		RowData should, RowData is, RowDataSerializer serializer1, RowDataSerializer serializer2) {
+			RowData should,
+			RowData is,
+			RowDataSerializer serializer1,
+			RowDataSerializer serializer2) {
+		return deepEqualsRowData(should, is, serializer1, serializer2, false);
+	}
+
+	private static boolean deepEqualsRowData(
+			RowData should,
+			RowData is,
+			RowDataSerializer serializer1,
+			RowDataSerializer serializer2,
+			boolean checkClass) {
 		if (should.getArity() != is.getArity()) {
 			return false;
 		}
+		if (checkClass && should.getClass() != is.getClass()) {
+			return false;
+		}
+
 		BinaryRowData row1 = serializer1.toBinaryRow(should);
```

Comment:
It always convert rows to `BinaryRowData` for comparing. Could we additionally compare the `RowData` directly when `checkClass` is true?